docs/madr/decisions/070-resource-identifier.md

```diff
@@ -0,0 +1,363 @@
+| VirtualHost       | legacy listeners - `<kuma.io/service>`<br>new outbounds - `<mesh>_<name>_<namespace>_<zone>_<short-name>_<port>`                                                                                        | Mesh*Service (with sectionName to select port) |
+| Inbound Cluster   | `localhost:<port>`                                                                                                                                                                                      | Dataplane (with sectionName to select port)    |
+| Outbound Cluster  | legacy clusters - `<kuma.io/service>-hash(dst.tags)`<br>legacy clusters cross-mesh - `<kuma.io/service>-hash(dst.tags)_<mesh>`<br>new clusters - `<mesh>_<name>_<namespace>_<zone>_<short-name>_<port>` | Mesh*Service (with sectionName to select port) |
+| Route             | Routes are set on Listener on VirtualHost.<br>On inbound - `inbound:<kuma.io/service>`<br>On outbound - `<hash_sha256([]Match{...})>`                                                                   | Correlates with a set of MeshHTTPRoutes        |
+| Secret            | `name       = <category>:<scope>:<identifier>`<br>`category   = "mesh_ca" \| "identity_cert"`<br>`scope      = "secret"`<br>`identifier = "all" \| <mesh_name>`                                         | TODO                                           |
+
+##### Builtin Gateway
+
+|             | Name                                                                                                                                                                                                                           | Correlated Resources                                  |
+|-------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------|
+| Listener    | `<gateway-name>:<protocol>:<port>` where `gateway-name` is `MeshGatewayResource.Meta.Name`                                                                                                                                     | MeshGateway (with sectionName to select the listener) |
+| VirtualHost | `<hostname>`                                                                                                                                                                                                                   |                                                       |
+| Cluster     | `<kuma.io/service>-hash(merge(dpp.tags, gateway.tags, listener.tags))`                                                                                                                                                         | Pair of MeshGateway and Mesh*Service                  |
```

Comment:
So what does this actually look like?

Reply:
Not much we can do right now as we generate a unique cluster per MeshGateway listener. I looked a bit into this and I think we should be able to stop doing that. I've created an issue https://github.com/kumahq/kuma/issues/13129 and referenced it in the MADR. Once we get rid of hash-suffixes in MeshGateway cluster names we can go with `kri` from Mesh*Service.